### PR TITLE
Zip docs organization

### PR DESCRIPTION
**Pull request by paulb777**
Split README.md into DEVELOP.md so that README.md is focused on users and DEVELOP.md is on development and Firebase release management.

Also, makes a few tweaks to the `--help` output.

### CODE REVIEW

File: ReleaseTooling/README.md [hunk 1]
```diff
@@ -59,93 +60,11 @@ Indicates to install the version 3.2.0 of "GoogleDataTransport" and the latest
 version of "FirebaseMessaging". The version string is optional and can be any
 valid [CocoaPods Podfile version specifier](https://guides.cocoapods.org/syntax/podfile.html#pod).
 
-
 Optional common arguments:
 - `--no-update-pod-repo`
   - This is for speedups when `pod repo update` has already been run recently.
```
Comment (paulb777) [inline ReleaseTooling/README.md:65]: See below the red for added optional arguments.

File: ReleaseTooling/Sources/ZipBuilder/main.swift [hunk 2]
```diff
@@ -102,14 +102,15 @@ struct ZipBuilderTool: ParsableCommand {
 
   /// The minimum iOS Version to build for.
   @Option(default: "10.0",
-          help: ArgumentHelp("The minimum supported iOS version. The default is 10.0."))
+          help: ArgumentHelp("The minimum supported iOS version."))
```
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/main.swift:105]: Redundant with automatic help already printing the default.

File: ReleaseTooling/Sources/ZipBuilder/main.swift [hunk 3]
```diff
@@ -102,14 +102,15 @@ struct ZipBuilderTool: ParsableCommand {
...
   var minimumIOSVersion: String
 
   /// The list of architectures to build for.
   @Option(parsing: .upToNextOption,
           help: ArgumentHelp("""
-          The list of platforms to build for. The default list is \
-          \(TargetPlatform.allCases.map { $0.sdkName }).
+          The list of target platforms to build for. The default list is \
+          \(TargetPlatform.allCases.map { $0.sdkName }). Note that `macosx` is currently Catalyst \
```
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/main.swift:112]: Output is 
`  --platforms <platforms> The list of target platforms to build for. The default list is ["iphoneos", "iphonesimulator", "macosx"]. Note that `macosx` is currently Catalyst only.`

The macosx was misleading before.

File: ReleaseTooling/DEVELOP.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+containing all the zip files and JSON files necessary for distribution.
+
+## Firebase Releaser
+
+Provides several functions for staging a Firebase release candidate. See the internal go/firi link
+for the process documentation.
+
+### Launch Arguments
+
+See `main.swift`  for information on specific launch arguments.
```
Comment (ryanwilson) [inline ReleaseTooling/DEVELOP.md:47]: Nit: extra space after closing `

Comment (paulb777) [reply ReleaseTooling/DEVELOP.md:47]: Done in upcoming commit in #7221

**Pull request merged**